src/server.ts

```diff
@@ -0,0 +1,93 @@
+import { Client } from "pg";
+import { config } from "dotenv";
+import express from "express";
+import cors from "cors";
+import filePath from "./filePath";
+
+config(); //Read .env file lines as though they were env vars.
+
+//Call this script with the environment variable LOCAL set if you want to connect to a local db (i.e. without SSL)
+//Do not set the environment variable LOCAL if you want to connect to a heroku DB.
+
+//For the ssl property of the DB connection config, use a value of...
+// false - when connecting to a local DB
+// { rejectUnauthorized: false } - when connecting to a heroku DB
+const herokuSSLSetting = { rejectUnauthorized: false };
+const sslSetting = process.env.LOCAL ? false : herokuSSLSetting;
+const dbConfig = {
+  connectionString: process.env.DATABASE_URL,
+  ssl: sslSetting,
+};
+
+const app = express();
+
+app.use(express.json()); //add body parser to each following route handler
+app.use(cors()); //add CORS support to each following route handler
+
+const client = new Client(dbConfig);
+client.connect();
+
+app.get("/", async (req, res) => {
+  const pathToFile = filePath("../public/index.html");
+  res.sendFile(pathToFile);
+});
+
+app.get("/breeds", async (req, res) => {
+  try {
+    const queryRes = await client.query(
+      "SELECT * FROM breedvotes ORDER BY vote DESC, dogbreed"
+    );
+    res.status(200).json(queryRes.rows);
+  } catch (error) {
+    res.status(404).send(error.stack);
+  }
+});
+
+app.post<{}, {}, { dogbreed: string }>("/breeds", async (req, res) => {
+  try {
+    const newBreed = req.body.dogbreed;
+    const query = `INSERT INTO breedvotes (dogbreed)  
+                VALUES ($1)
+                Returning *`;
+    const queryRes = await client.query(query, [newBreed]);
+    if (queryRes.rowCount === 0) {
+      res.status(400);
+    } else {
+      res.status(201).send(queryRes.rows);
+    }
+  } catch (error) {
+    res.status(404).send(error.stack);
+  }
+});
+
+app.put<{ id: string }, {}, { currentVote: number }>(
+  "/breeds/:id",
+  async (req, res) => {
+    try {
+      const breedId = parseInt(req.params.id);
+      const currentBreedVote = req.body.currentVote;
```

Comment:
Relies on the client having the most up-to-date value of vote count. If user hadn't refreshed and another casted a vote, the user would then be sending the server the wrong values for currentBreedVote

Reply:
Jacob, would love to have a further discussion about this as we are struggling to wrap our heads around the difference.